### PR TITLE
Added all-in-one example

### PR DESCRIPTION
**Pull request by twolfson**
This is 1 of 2 PRs for #40. We decided to break them up into separate PRs to prevent overwhelming the reviewer. Although, we should consider a 3rd PR which unifies duplicate content somehow.

In this PR:

- Added `examples/all_in_one` (casing consistent with rest of repo)
- Explanation available in `README.rst`
    - https://github.com/twolfson/sops/tree/dev/add.examples.sqwished/examples/all_in_one

### CODE REVIEW

File: examples/all_in_one/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+config/secret.json
```
Comment (jvehent) [inline examples/all_in_one/.gitignore]: what's the purpose of having a .gitignore file here?

File: examples/all_in_one/README.rst [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    # From the `sops` root directory
+    # Import the test key
+    gpg --import tests/sops_functional_tests_key.asc
+
+    # Navigate to our example directory
+    cd examples/all_in_one
+
+    # Decrypt our secrets
+    bin/decrypt-secrets.sh
```
Comment (jvehent) [inline examples/all_in_one/README.rst]: I think you mean `decrypt-config.sh`

File: examples/all_in_one/README.rst [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    # Optionally edit a secret
+    # bin/edit-secret.sh config/secret.enc.json
+
+    # Run our script
+    python main.py
```
Comment (jvehent) [inline examples/all_in_one/README.rst]: This is a bit obscure, what is it meant to do?

**Comment by jvehent** [comment]
This looks alright except for a couple minor comments. Maybe describing the intend at the top would help the reader follow along a bit more?

File: examples/all_in_one/README.rst (hunk 2 above)
Comment (twolfson) [inline examples/all_in_one/README.rst]: Ah, nice catch

File: examples/all_in_one/README.rst (hunk 3 above)
Comment (twolfson) [inline examples/all_in_one/README.rst]: It's means to be an example script that relies on the config (e.g. running an application). I will update the comment to make it more clear

**Comment by twolfson** [comment]
I will see what I can do. I will take care of things by the end of this weekend and leave a comment when it's updated

**Comment by twolfson** [comment]
We have addressed the comments and updated the PR

**Comment by jvehent** [comment]
This looks good. Thanks a lot for the PR!

**Pull request merged**